### ISSUE TITLE
pgtap-core--1.0.0.sql:915: ERROR:  column p.proisagg does not exist

### ISSUE THREAD

**Issue by cbandy** [theory/pgtap#197]
When building and loading `pgtap-core--1.0.0.sql` in PostgreSQL 11, some pre-11 columns are referenced:

```
psql:pgtap-1.0.0/sql/pgtap-core--1.0.0.sql:915: ERROR:  column p.proisagg does not exist
LINE 11:            p.proisagg    AS is_agg,
                    ^
HINT:  Perhaps you meant to reference the column "p.prolang".
```

### Steps to reproduce:

1. Run PostgreSQL 11, in a container here:
```
$ docker run --rm -it docker.io/library/buildpack-deps:stretch

# curl -sS https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
# echo > /etc/apt/sources.list.d/pgdg.list 'deb http://apt.postgresql.org/pub/repos/apt/ stretch-pgdg main'
# apt-get update && apt-get install --no-install-recommends --yes postgresql-11 postgresql-server-dev-11
# service postgresql start

# su -c 'psql -c "select version()"' postgres
                                                             version                                                              
----------------------------------------------------------------------------------------------------------------------------------
 PostgreSQL 11.2 (Debian 11.2-1.pgdg90+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516, 64-bit
(1 row)
```

2. Build PgTAP 1.0.0:
```
# curl -L 'https://github.com/theory/pgtap/archive/v1.0.0.tar.gz' | tar xz

# make -C pgtap-1.0.0 all
make: Entering directory '/pgtap-1.0.0'                                                                                          
Makefile:86: To use pg_prove, TAP::Parser::SourceHandler::pgTAP Perl module
Makefile:87: must be installed from CPAN. To do so, simply run:   
Makefile:88: cpan TAP::Parser::SourceHandler::pgTAP
cp sql/pgtap--0.96.0--0.97.0.sql.in sql/pgtap--0.96.0--0.97.0.sql                                                               
cp sql/pgtap--0.97.0--0.98.0.sql.in sql/pgtap--0.97.0--0.98.0.sql
cp sql/pgtap--0.95.0--0.96.0.sql.in sql/pgtap--0.95.0--0.96.0.sql
cp sql/pgtap--0.99.0--1.0.0.sql.in sql/pgtap--0.99.0--1.0.0.sql     
cp sql/pgtap.sql.in sql/pgtap.sql                                                                                                 
sed -e 's,MODULE_PATHNAME,$libdir/pgtap,g' -e 's,__OS__,linux,g' -e 's,__VERSION__,1.0,g' sql/pgtap.sql > sql/pgtap.tmp          
mv sql/pgtap.tmp sql/pgtap.sql
grep '^CREATE ' sql/pgtap.sql | '/usr/bin/perl' -e 'for (reverse <STDIN>) { chomp; s/CREATE (OR REPLACE )?/DROP /; print "$_;\n" }' | sed 's/DROP \(FUNCTION\|VIEW\|TYPE\)
 /DROP \1 IF EXISTS /' > sql/uninstall_pgtap.sql                                                          
cp sql/pgtap.sql.in sql/pgtap-static.sql                                                
sed -e 's,sql/pgtap,sql/pgtap-static,g' compat/install-10.patch | patch -p0
patching file sql/pgtap-static.sql
sed -e 's,sql/pgtap,sql/pgtap-static,g' compat/install-9.6.patch | patch -p0
patching file sql/pgtap-static.sql
sed -e 's,sql/pgtap,sql/pgtap-static,g' compat/install-9.4.patch | patch -p0
patching file sql/pgtap-static.sql
sed -e 's,sql/pgtap,sql/pgtap-static,g' compat/install-9.2.patch | patch -p0
patching file sql/pgtap-static.sql
sed -e 's,sql/pgtap,sql/pgtap-static,g' compat/install-9.1.patch | patch -p0
patching file sql/pgtap-static.sql
sed -e 's,sql/pgtap,sql/pgtap-static,g' compat/install-9.0.patch | patch -p0
patching file sql/pgtap-static.sql
sed -e 's,sql/pgtap,sql/pgtap-static,g' compat/install-8.4.patch | patch -p0
patching file sql/pgtap-static.sql
sed -e 's,sql/pgtap,sql/pgtap-static,g' compat/install-8.3.patch | patch -p0
patching file sql/pgtap-static.sql
sed -e 's,MODULE_PATHNAME,$libdir/pgtap,g' -e 's,__OS__,linux,g' -e 's,__VERSION__,1.0,g' sql/pgtap-static.sql > sql/pgtap-static.tmp
mv sql/pgtap-static.tmp sql/pgtap-static.sql
'/usr/bin/perl' compat/gencore 0 sql/pgtap-static.sql > sql/pgtap-core.sql
'/usr/bin/perl' compat/gencore 1 sql/pgtap-static.sql > sql/pgtap-schema.sql
cp sql/pgtap.sql sql/pgtap--1.0.0.sql
cp sql/pgtap-core.sql sql/pgtap-core--1.0.0.sql
cp sql/pgtap-schema.sql sql/pgtap-schema--1.0.0.sql
make: Leaving directory '/pgtap-1.0.0'
```

3. Run pgtap-core.sql:
```
# su -c 'psql -qx -v ON_ERROR_STOP=1 -f pgtap-1.0.0/sql/pgtap-core--1.0.0.sql' postgres
psql:pgtap-1.0.0/sql/pgtap-core--1.0.0.sql:915: ERROR:  column p.proisagg does not exist
LINE 11:            p.proisagg    AS is_agg,
                    ^
HINT:  Perhaps you meant to reference the column "p.prolang".
```

**Comment by theory** [comment]
Ugh, pgtap-core is supposed to work the same, without modification, on any version of Postgres. The need to patch this bit depending on the version will be a PITA.

**Comment by myprak** [comment]
I got it working by replacing the line with

`CASE WHEN p.prokind='a' THEN TRUE ELSE FALSE END AS is_agg,`

Yes, this is only a Postgres 11 patch. Lots of annoyance on the Pg boards about non backward compatible change

**Comment by nasbyj** [comment]
Fixed in master / 1.1.0
